source/posix/socket.c

```diff
@@ -1263,7 +1268,10 @@ static void s_close_task(struct aws_task *task, void *arg, enum aws_task_status
 
 int aws_socket_close(struct aws_socket *socket) {
     struct posix_socket *socket_impl = socket->impl;
-    AWS_LOGF_DEBUG(AWS_LS_IO_SOCKET, "id=%p fd=%d: closing", (void *)socket, socket->io_handle.data.fd);
+    int fd = socket->io_handle.data.fd; /* cached for logging */
+    (void)fd;
```

Comment:
Trivial: I am not getting the idea of caching... Will it be overwritten later by eventloop thread?🤔

Reply:
yeah, there's a branch here where cleanup is performed on another thread while this thread waits. Then this thread logs that it's done, but the log statement was like:
`fd=7 closing from another thread`
and the next log statement was:
`fd=-1 close task completed`

I'll fix the name/comments here to be more clear